Detectors/EMCAL/base/src/ClusterFactory.cxx

```diff
@@ -551,24 +555,145 @@ void ClusterFactory<InputType>::evalElipsAxis(gsl::span<const int> inputsIndices
 /// Finds the maximum energy in the cluster and computes the Summed amplitude of digits/cells
 //____________________________________________________________________________
 template <class InputType>
-std::tuple<int, float, float> ClusterFactory<InputType>::getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const
+std::tuple<int, float, float, bool> ClusterFactory<InputType>::getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const
 {
 
   float energy = 0.;
   int mid = 0;
   float cellAmp = 0;
+  int iSupMod0 = -1;
+  bool shared = false;
   for (auto iInput : inputsIndices) {
     if (iInput >= mInputsContainer.size()) {
       throw CellIndexRangeException(iInput, mInputsContainer.size());
     }
     cellAmp += mInputsContainer[iInput].getEnergy();
+    if (iSupMod0 == -1) {
+      iSupMod0 = mGeomPtr->GetSuperModuleNumber(mInputsContainer[iInput].getTower());
+    } else if (iSupMod0 != mGeomPtr->GetSuperModuleNumber(mInputsContainer[iInput].getTower())) {
+      shared = true;
+    }
     if (mInputsContainer[iInput].getEnergy() > energy) {
       energy = mInputsContainer[iInput].getEnergy();
       mid = iInput;
     }
   } // loop on cluster inputs
 
-  return std::make_tuple(mid, energy, cellAmp);
+  return std::make_tuple(mid, energy, cellAmp, shared);
+}
+
+///
+/// Look to cell neighbourhood and reject if it seems exotic
+//____________________________________________________________________________
+template <class InputType>
+bool ClusterFactory<InputType>::isExoticCell(short towerId, float ecell) const
+{
+  if (ecell < mExoticCellMinAmplitude) {
+    return false; // do not reject low energy cells
+  }
+
+  float eCross = getECross(towerId, ecell);
+
+  if (1 - eCross / ecell > mExoticCellFraction) {
+    LOG(debug) << Form("EXOTIC CELL id %d, eCell %f, eCross %f, 1-eCross/eCell %f",
+                       towerId, ecell, eCross, 1 - eCross / ecell);
```

Comment:
Should not hold up the merging but for logging you should either use the stream operators directly or in case you want a format string `LOGF`. Using ROOTs `Form` together with a stream does not make too much sense, neither is it very performant. Note that in o2 we rather use the fmt library in case we would need format strings.